agent-inject/agent/annotations.go

```diff
@@ -20,18 +20,41 @@ const (
 	// be set to a true or false value, as parseable by strconv.ParseBool
 	AnnotationAgentInject = "vault.hashicorp.com/agent-inject"
 
-	// AnnotationAgentInjectSecret is the key annotation that configures Vault
+	// (Deprecated) AnnotationAgentInjectSecret is the key annotation that configures Vault
```

Comment:
Since we are deprecating this, why not store json as the value of annotation, that way we can have more structured data when we process and also it would require less number of annotations. 

Example

``` yaml
vault.hashicorp.com/agent-inject-secret-config-FOOBAR: '{"secretPath": "secrets/service/database", "fileName": "database", "mountPath": "/etc/container_environment", "template": "<template>", "postInjectionCommand": "<optional command>"}'

PS: we can come up with better names 
```

Reply:
I can easily make that change but would like some guidance from the doce owners here. If we are to move to one annotation with metadata, should we still also support the old annotations? (I think not)
If we want to support the legacy annotations, and both are present, which one should take precedence?

Reply:
IMO new one should take precedence and we should support old one for backwards comparability and print warning in logs everytime it’s used

Reply:
Throwing error incase both are present can also be an option. There is something definitely wrong if both are present.